core/lib/dal/src/pruning_dal.rs

```diff
@@ -0,0 +1,345 @@
+                    FROM
+                        storage_logs
+                    WHERE
+                        miniblock_number BETWEEN $1 AND $2
+                    GROUP BY
+                        hashed_key
+                ) AS last_storage_logs
+                WHERE
+                    storage_logs.miniblock_number BETWEEN $1 AND $2
+                    AND last_storage_logs.hashed_key = storage_logs.hashed_key
+                    AND (
+                        storage_logs.miniblock_number != last_storage_logs.op[1]
+                        OR storage_logs.operation_number != last_storage_logs.op[2]
+                    )
+                "#,
+                i64::from(first_miniblock_to_prune.0),
+                i64::from(last_miniblock_to_prune.0),
+            )
+            .instrument(
+                "hard_prune_batches_range#delete_overriden_storage_logs_from_pruned_batches",
+            )
+            .with_arg("first_miniblock_to_prune", &first_miniblock_to_prune)
+            .with_arg("last_miniblock_to_prune", &last_miniblock_to_prune)
+            .report_latency()
+            .execute(self.storage)
+            .await?;
+
+            sqlx::query!(
+                r#"
+                DELETE FROM l1_batches
+                WHERE
+                    number <= $1
+                "#,
+                i64::from(last_l1_batch_to_prune.0),
+            )
+            .instrument("hard_prune_batches_range#delete_l1_batches")
+            .with_arg("last_l1_batch_to_prune", &last_l1_batch_to_prune)
+            .report_latency()
+            .execute(self.storage)
+            .await?;
+
+            sqlx::query!(
+                r#"
+                DELETE FROM miniblocks
+                WHERE
+                    number <= $1
+                "#,
+                i64::from(last_miniblock_to_prune.0),
+            )
+            .instrument("hard_prune_batches_range#delete_miniblocks")
+            .with_arg("last_l1_batch_to_prune", &last_l1_batch_to_prune)
+            .report_latency()
+            .execute(self.storage)
+            .await?;
+        }
+
+        sqlx::query!(
+            r#"
+            INSERT INTO
+                pruning_log (
+                    pruned_l1_batch,
+                    pruned_miniblock,
+                    TYPE,
+                    created_at,
+                    updated_at
+                )
+            VALUES
+                ($1, $2, $3, NOW(), NOW())
+            "#,
+            i64::from(last_l1_batch_to_prune.0),
+            i64::from(last_miniblock_to_prune.0),
+            PruneType::Hard as PruneType
+        )
+        .instrument("soft_prune_batches_range#insert_pruning_log")
+        .with_arg("last_l1_batch_to_prune", &last_l1_batch_to_prune)
+        .with_arg("last_miniblock_to_prune", &last_miniblock_to_prune)
+        .with_arg("prune_type", &PruneType::Hard)
+        .report_latency()
+        .execute(self.storage)
+        .await?;
```

Comment:
I think it would be useful to add detailed logging for this. That is - how many entities of each type were deleted

Reply:
corrected